lib/synapse/config_generator/haproxy.rb

```diff
@@ -961,17 +973,28 @@ def validate_haproxy_stanza(stanza, stanza_type, service_name)
 
     # generates an individual stanza for a particular watcher
     def generate_frontend_stanza(watcher, config)
-      unless watcher.haproxy.has_key?("port")
+      unless watcher.generator_config['haproxy'].has_key?("port")
```

Comment:
Sure, I already did this for nginx, but was punting on this one. I can make the changes now though.